uqcsbot/api.py

```diff
@@ -224,6 +232,19 @@ def _initialise(self):
                     self._add_channel(im)
             self._initialised = True
 
+    def populate_from_team_state(self, data):
+        with self._lock:
+            self._initialised = True
+            self._channels_by_id = {}
+            self._channels_by_name = {}
+            chan_types = ['channels', 'groups', 'ims']
```

Comment:
just inline this

Reply:
Gonna maybe make a base class for this shit seeing it's basically the same on users and channels

Reply:
OK I'm not gonna make a base class for this